nestkernel/recording_backend_sionlib.h

```diff
@@ -75,12 +79,39 @@ filename is composed.
 Data format
 +++++++++++
 
-The container files contain the data in a custom binary format. The
-recorder property ``label`` and all other recorder-specific metadata
-is stored with each recording device. To retrieve the data after the
-simulation, the `nestio reader API` in Python has to be used. SIONlib
-container files are sparse files: They only use up the space on disk
-which is actually filled with data.
+In contrast to other recording backends, the ``sionlib`` backend
+writes the data from all recorders using it to a single container
+file(s). The file(s) contain the data in a custom binary format, which
+is composed of a series of blocks in the following order:
+
+* the *body block* contains the actual data records; the layout of an
+  individual record depends on the type of the device and is described
+  by a corresponding entry in the *device info block*
+  
+* the *file info block* keeps the file's metadata like version
+  information and such.
+
+* the *device info block* stores the properties and a data layout
+  description for each device that uses the ``sionlib`` backend.
+
+* the *tail block* contains pointers to the *file info block*
+
+The data layout of the NEST SIONlib file format v2 is shown in the
+following figure.
+
+.. figure:: ../_static/img/nest_sionlib_file_format_v2.png
+   :alt: NEST SIONlib binary file format
+
+   NEST SIONlib binary file format.
+
+
+Reading the data
+++++++++++++++++
+
+As the binary format of the files produced by the ``sionlib`` backend,
+we provide a custom reader module for Python that makes the files
+available in a convenient way. The source code for this can be found
```

Comment:
Is there something missing in the first part of this sentence? Right now it seems to imply that we are the binary format, which is a bit too existential for this documentation, I think.

Reply:
Fixed by 169047c47.